### DISCUSSION TITLE
oAuth redirection based on Shopify app requirments

### DISCUSSION THREAD

**Discussion by APdevelopments** [Kyon147/laravel-shopify#257] [General]
I'm using an ancient version of the package and Laravel 5.8. I cant change to newer because of some must-have libraries that do not work with the newest version of Laravel.

* Package Version: v12.1
* Laravel Version: v5.8
* PHP Version: v7.3

I got a URGENT email from Shopify, and they delisted my app because of: '**A. Authentication[Anchor link to section titled "A. Authentication"](https://shopify.dev/docs/apps/store/requirements#a-authentication)
Your app must immediately authenticate using OAuth before any other steps occur, even if the merchant has previously installed and then uninstalled your app."**

I looked at my version (12.1), and AuthController.php, AuthenticateShop.php, and AuthorizeShop.php handle that part.
Currently,/authenticate redirects the user to the authenticate/oauth?shop=exampleshop.myshopify.com, and that is a problem for Shopify, and they delisted my app.

Please help me fix the issue as soon as possible because it's critical for my business.
What I have to change in these files to redirect users immediately to Shopify oAuth page. I know I can use the latest library version, but unfortunately, it's not possible in my case.

**Comment by APdevelopments** [comment]
Very urgent if someone can help

**Comment by Kyon147** [comment]
@APdevelopments I'm moving this to a discussion as it is not an issue with the package.

**Comment by Kyon147** [comment]
Without looking at your app directly, the advice I'd give you is to see if you can update this package and then change your other packages or re-write that part of the app which uses those older packages.

It will be more messy to retro-fit 8-9 version of updates from our package into an older version, that probably update your business logic.
